lucene/core/src/test/org/apache/lucene/search/TestWANDScorer.java

```diff
@@ -947,4 +988,82 @@ public long cost() {
       };
     }
   }
+
+  private static class WANDScorerQuery extends Query {
+    private final BooleanQuery query;
+
+    private WANDScorerQuery(BooleanQuery query) {
```

Comment:
I wonder if it would make the tests easier to read if we took an array of queries here:

```suggestion
    private WANDScorerQuery(Query... query) {
```

while still creating a `BooleanQuery` under the hood to reuse equals/hashcode/etc.

Reply:
Hmm do you mean for current caller pattern like the following:

```
Query query =
            new BooleanQuery.Builder()
                .add(
                    new WANDScorerQuery(
                        new BooleanQuery.Builder()
                            .add(
                                new BoostQuery(
                                    new ConstantScoreQuery(new TermQuery(new Term("foo", "A"))), 2),
                                Occur.SHOULD)
                            .add(
                                new ConstantScoreQuery(new TermQuery(new Term("foo", "B"))),
                                Occur.SHOULD)
                            .add(
                                new BoostQuery(
                                    new ConstantScoreQuery(new TermQuery(new Term("foo", "D"))), 4),
                                Occur.SHOULD)
                            .setMinimumNumberShouldMatch(2)
                            .build()),
                    Occur.MUST)
                .add(new TermQuery(new Term("foo", "C")), Occur.FILTER)
                .build();
```

we would like to see if we can create the same via the following calling pattern:

```
// missing Occur.MUST from the above between boolean queries
Query = new WANDScorerQuery(
                        new BooleanQuery.Builder()
                            .add(
                                new BoostQuery(
                                    new ConstantScoreQuery(new TermQuery(new Term("foo", "A"))), 2),
                                Occur.SHOULD)
                            .add(
                                new ConstantScoreQuery(new TermQuery(new Term("foo", "B"))),
                                Occur.SHOULD)
                            .add(
                                new BoostQuery(
                                    new ConstantScoreQuery(new TermQuery(new Term("foo", "D"))), 4),
                                Occur.SHOULD)
                            .setMinimumNumberShouldMatch(2)
                            .build(),

                        new BooleanQuery.Builder().add(new TermQuery(new Term("foo", "C")), Occur.FILTER).build());
```

I feel this constructor API may not be easy to capture the (boolean) relationship between queries, and also allow non boolean queries to be passed in, which may complicate logic inside the class?

Reply:
I was thinking of something like that indeed.